server/src/main/java/org/opensearch/search/fetch/subphase/InnerHitsPhase.java

```diff
@@ -64,7 +64,7 @@ public InnerHitsPhase(FetchPhase fetchPhase) {
 
     @Override
     public FetchSubPhaseProcessor getProcessor(FetchContext searchContext) {
-        if (searchContext.hasInnerHits() == false) {
+        if (searchContext.innerHits() == null) {
             return null;
         }
```

Comment:
We can remove this check as the `return null` is never hit! Probably add some documentation to the effect that we need innerHitsProcessor always included